### ISSUE TITLE
Duplicate parameters don't produce an error

### ISSUE THREAD

**Issue by joshuaspence** [hashicorp/terraform#13280]
It seems that duplicate resource parameters don't cause an error or even a warning. Using the latest Terraform release (0.9.2), consider the following code:

```
resource "random_id" "test" {
  byte_length = 1
  byte_length = 2
}
```

Running `terraform plan` produces no errors or warnings and seems to choose the last value specified for `byte_length`. This seems like a bug as it makes it more difficult to debug the behavior of Terraform.

```
+ random_id.test
    b64:         "<computed>"
    b64_std:     "<computed>"
    b64_url:     "<computed>"
    byte_length: "2"
    dec:         "<computed>"
    hex:         "<computed>"


Plan: 1 to add, 0 to change, 0 to destroy.
```

**Comment by joshuaspence** [comment]
I guess `variable`s and `output`s should also produce a warning or error.

**Comment by apparentlymart** [comment]
Hi @joshuaspence!

I just confirmed that this is resolved in Terraform v0.12.0-alpha2:

![Error: Attribute redefined on duplicate-args.tf line 3](https://user-images.githubusercontent.com/20180/48743660-d7482580-ec18-11e8-9086-e54adc45c8ed.png)


There are similar checks for duplicate declarations of variables, outputs, resources, etc.

Thanks for reporting this, and sorry it took so long to fix it!
